### DISCUSSION TITLE
Can I move focus and swap windows from the command line?

### DISCUSSION THREAD

**Discussion by WJCFerguson** [paperwm/PaperWM#851] [Q&A] [answered]
I (believe I) want to be able to trigger directional activation and movement of windows from a script or command.  Is this possible?

As a secondary issue I want to get the title of the focused window, but I know I can do that with external tools if necessary.

My specific usage: I use Emacs, with its own internal "window" (aka "pane") management.  I bind keys to a script that passes the request to Emacs if it's focused but otherwise passes them on to the WM.  Also when I hit the edge of Emacs I pass the request out to the WM.  I've done this in `i3` and `herbstluftwm` and others and at this point I don't want to have to live without it.  e.g. in i3 I can run `i3-msg focus left`.

**Comment by Lythenas** [comment, marked as answer]
At the moment this is not possible. And unfortunately (as far as I know) Gnome doesn't really provide an easy to use mechanism to do it.

I have been experimenting a bit with DBus. See #774 That would allow you to do what you want. But it's still just a proof of concept.

**Comment by WJCFerguson** [comment]
I think my workaround may be to set up obscure bindings I'll never manually type, and having `ydotool` send them when I want to script actions.  e.g. bindings with `Super-Alt-Control-Shift`.
